From 9a8fc811282e303ec2f93c978228a85961dae263 Mon Sep 17 00:00:00 2001
From: Noah Davis <noahadvs@gmail.com>
Date: Mon, 14 Sep 2020 10:18:57 -0400
Subject: [PATCH] [sddm-theme/lockscreen] Fix login button size

Some users reported this button becoming too small after KF5.74 was released, but I could not reproduce it myself. However, this code did look suspicious. Hopefully this fixes the issue.
---
 lookandfeel/contents/lockscreen/MainBlock.qml | 4 ++--
 sddm-theme/Login.qml                          | 5 ++---
 2 files changed, 4 insertions(+), 5 deletions(-)

--- a/lookandfeel/contents/lockscreen/MainBlock.qml
+++ b/lookandfeel/contents/lockscreen/MainBlock.qml
@@ -97,7 +97,8 @@
         PlasmaComponents.Button {
             id: loginButton
             Accessible.name: i18nd("plasma_lookandfeel_org.kde.lookandfeel", "Unlock")
-            implicitHeight: passwordBox.height - units.smallSpacing * 0.5 // otherwise it comes out taller than the password field
+            Layout.preferredHeight: passwordBox.implicitHeight
+            Layout.preferredWidth: loginButton.Layout.preferredHeight
 
             PlasmaCore.IconItem { // no iconSource because if you take away half a unit (implicitHeight), "go-next" gets cut off
                     anchors.fill: parent
--- a/sddm-theme/Login.qml
+++ b/sddm-theme/Login.qml
@@ -112,8 +112,8 @@
         PlasmaComponents.Button {
             id: loginButton
             Accessible.name: i18nd("plasma_lookandfeel_org.kde.lookandfeel", "Log In")
-            implicitHeight: passwordBox.height - units.smallSpacing * 0.5 // otherwise it comes out taller than the password field
-            Layout.rightMargin: 1 // prevents it from extending beyond the username field
+            Layout.preferredHeight: passwordBox.implicitHeight
+            Layout.preferredWidth: loginButton.Layout.preferredHeight
 
             PlasmaCore.IconItem { // no iconSource because if you take away half a unit (implicitHeight), "go-next" gets cut off
                 anchors.fill: parent
